src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (
+      !(animal instanceof Carnivore) &&
+      animal instanceof Herbivore &&
+      animal.hidden === false
```

Comment:
The check for `!(animal instanceof Carnivore)` is redundant since you are already checking if the animal is an instance of `Herbivore` and not hidden. You can remove this check to simplify the condition.